web-wallet/src/lib/wallet-cache/__tests__/index.spec.js

```diff
@@ -264,24 +294,121 @@ describe("Wallet cache", () => {
         unspentNotesToAdd
       );
 
-      await walletCache.addUnspentNotes(newNotes);
+      await walletCache.addUnspentNotes(newNotes, newSyncInfo);
 
       await expect(
         walletCache.getUnspentNotes().then(sortByNullifier)
       ).resolves.toStrictEqual(sortByNullifier(expectedUnspentNotes));
+      await expect(walletCache.getSyncInfo()).resolves.toStrictEqual(
+        newSyncInfo
+      );
     });
 
-    it("should leave the unspent notes as they were if an error occurs during insertion", async () => {
+    it("should leave the unspent notes and the sync info as they were if an error occurs during insertion", async () => {
       // @ts-expect-error
       const newNotes = cacheSpentNotes.concat({});
 
       await expect(
-        walletCache.addUnspentNotes(newNotes)
+        walletCache.addUnspentNotes(newNotes, newSyncInfo)
       ).rejects.toBeInstanceOf(Error);
 
       expect(sortByNullifier(cacheUnspentNotes)).toStrictEqual(
         sortByNullifier(await walletCache.getUnspentNotes())
       );
+      await expect(walletCache.getSyncInfo()).resolves.toStrictEqual(
+        currentSyncInfo
+      );
+    });
+  });
+
+  describe("Spending notes", () => {
+    it("should expose a method to move a group of notes from the unspent to the spent table", async () => {
+      /** @type {(entry: { nullifier: Uint8Array }) => Uint8Array} */
+      const getNullifier = getKey("nullifier");
+      const [pendingToSpend, expectedPending] = await walletCache
+        .getPendingNotesInfo()
```

Comment:
I guess it's the expected behavior, as it looks like it is, based on the tests, but still wanted to mention it:

If `nullifiers` is `undefined`, `getPendingNotesInfo` will fetch all pending notes. I am wondering if there is a possible case where the `getPendingNotesInfo` will be called with `undefined`, when it shouldn't be.